site/blog/tutorials/iasql-on-ec2.mdx

```diff
@@ -0,0 +1,284 @@
+```sql
+SELECT * FROM key_pair_import('iasql', '<your-public-key-content>', 'us-east-1');
+```
+
+This will direct IaSQL to create an [AWS key pair](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/create-key-pairs.html#how-to-generate-your-own-key-and-import-it-to-aws) with the name `iasql` in the `us-east-1` region, and with your public key contents.
+
+:::note
+An IaSQL RPC is a function available in Postgres that upon calling, sends a request to the IaSQL engine to run a pre-defined logic and returns results to the Postgres console. In the above example [`key_pair_import`](https://iasql.com/docs/modules/aws/tables/aws_ec2_rpcs_import.KeyPairImportRpc/) is an IaSQL RPC that triggers the IaSQL engine to send a request to AWS for importing the public key.
+:::
+
+#### Create And Configure Security Group
+
+Now it's time to launch our EC2 instance. Before we go for the instance itself, let's first define the security group that's going to be attached to it. For this purpose, we'll need [`aws_security_group`](https://iasql.com/docs/modules/aws/aws_security_group/) module. But since `aws_ec2` module depends on it, it's already installed because IaSQL automatically installs the dependencies for modules upon an `iasql_install` call.
+
+<ThemedImage
+  alt="aws_ec2 module dependencies"
+  style={{width: '440'}}
+  sources={{
+    light: useBaseUrl('/img/module-dependency.png'),
+    dark: useBaseUrl('/img/module-dependency_dark.png'),
+  }}
+/>
+
+
+```sql
+INSERT INTO security_group (description, group_name)
+VALUES ('IaSQL security group', 'iasql_sg');
+
+WITH sg as (SELECT id
+            FROM security_group
+            WHERE group_name = 'iasql_sg')
+INSERT
+INTO security_group_rule (is_egress, ip_protocol, from_port, to_port, cidr_ipv4, description, security_group_id)
+VALUES (FALSE, 'tcp', 22, 22, '0.0.0.0/0', 'ssh port is accessible from all ips', (SELECT id FROM sg)),
+       (FALSE, 'tcp', 9879, 9879, '<your-ip>/0', 'open dashboard to just me', (SELECT id FROM sg)),
+       (FALSE, 'tcp', 5432, 5432, '0.0.0.0/0', 'postgres port is accessible from all ips', (SELECT id FROM sg)),
+       (TRUE, 'tcp', 1, 65535, '0.0.0.0/0', 'allow outgoing traffic to anywhere', (SELECT id FROM sg));
+```
+The first query simply creates a security group named `iasql_sg`. The second query uses a [`WITH` Postgres query](https://www.postgresql.org/docs/current/queries-with.html) to select the `id` of our security group to be used later in the `INSERT` query. The `WITH` query can be though of as a temporary table that exists just for one query. By use of `WITH` we can simplify our query and preload the `id` value from the `security_group` table. We could've written the query using [subqueries](https://www.postgresql.org/docs/current/functions-subquery.html) as follows, but that way we should've repeated the `SELECT id FROM security_group WHERE ...` part for each set of values:
```

Comment:
i'm not sure if explaining the syntax of the query and reasoning for using WITH is in the scope of this post. But open to other opinions

Reply:
I think this post already has a lot of things that are not directly the scope of it :D. Like discussing `cloud-init` directives. Do you think it should solely talk about the deployment, and not explain the concepts that are mentioned?

Reply:
yes, I think it will depend on how much we want to explain SQL concepts or assume it is known by the reader.